src/gpu/multiexp/multiexp.cl

```diff
@@ -19,24 +19,26 @@ __kernel void POINT_bellman_multiexp(
     uint window_size) {
 
   // We have `num_windows` * `num_groups` threads per multiexp.
-  uint gid = get_global_id(0);
+  const uint gid = get_global_id(0);
   if(gid >= num_windows * num_groups) return;
 
   // We have (2^window_size - 1) buckets.
-  uint bucket_len = ((1 << window_size) - 1);
+  const uint bucket_len = ((1 << window_size) - 1);
 
   // Each thread has its own set of buckets in global memory.
   buckets += bucket_len * gid;
-  for(uint i = 0; i < bucket_len; i++) buckets[i] = POINT_ZERO;
 
-  uint len = (uint)ceil(n / (float)num_groups); // Num of elements in each group
+  const POINT_projective tmp = POINT_ZERO;
```

Comment:
Maybe change `tmp` to `local_zero` or something like that

Reply:
done